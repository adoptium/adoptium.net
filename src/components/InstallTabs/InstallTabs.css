.install .red {
  background-color: #fa6157;
}

.install .yellow {
  background-color: #fbbc30;
}

.install .green {
  background-color: #2cc741;
}

.install .react-tabs {
  -webkit-tap-highlight-color: transparent;
}

.install .react-tabs__tab-list {
  display: flex;
  margin: 0;
  padding: 0;
  text-align: center;
}

.install .react-tabs__tab {
  background: #515c62;
  border: 1px solid transparent;
  border-bottom: none;
  bottom: -1px;
  color: #cbd4d9;
  cursor: pointer;
  display: inline-block;
  list-style: none;
  padding: 10px 12px;
  position: relative;
  text-align: center;
  width: 100%;
}

.install .react-tabs__tab--selected {
  background: #1f272a;
  border-radius: 5px 5px 0 0;
  color: #fff;
}

.install .react-tabs__tab:focus {
  border-bottom: none;
  border-color: #ed5393;
  box-shadow: 0 0 5px #ed5393;
  outline: none;
}

.install .react-tabs__tab:focus::after {
  background: #1f272a;
  bottom: -5px;
  content: "";
  height: 5px;
  left: 0;
  position: absolute;
  right: 0;
}

.install .react-tabs__tab-panel {
  display: none;
  padding: 20px;
}

.install .react-tabs__tab-panel--selected {
  display: block;
}

.install__header {
  height: 30px;
  margin: 0 5px;
}

.install__header-circles {
  float: left;
  height: 100%;
}

.install__header-grey-circle {
  background-color: #556066;
  border-radius: 50%;
  display: inline-block;
  height: 10px;
  margin: 10px 5px;
  width: 10px;
}

.install__header-text {
  line-height: 30px;
  text-align: center;
  width: calc(100% - 60px);
}

.install__text {
  color: #ed5393;
}

.install__text__line {
  margin: 1.2rem 0;
}

.install__text__no-select {
  margin-right: 1.2rem;
  user-select: none;
}

.install__docs-button,
.install__docs-button:hover {
  border: 1px solid #d9e1e4;
  border-radius: 0.4rem;
  bottom: 2rem;
  box-sizing: border-box;
  color: #d9e1e4;
  height: 2.8rem;
  padding: 0.8rem;
  position: absolute;
  right: 2rem;
  text-decoration: none;
  margin-right: 2em;
}
