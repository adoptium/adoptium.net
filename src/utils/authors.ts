import AuthorData from "@/data/authors.json";

export interface Author {
  name: string;
  summary?: string;
  twitter?: string;
  github?: string;
  linkedin?: string;
}

export function getFormattedAuthorData(authorId: string): Author {
  const rawAuthorData = AuthorData[authorId as keyof typeof AuthorData];

  if (!rawAuthorData) {
    return {
      name: "Unknown Author",
    };
  }

  // Convert the raw author data to our Author interface
  return {
    name: rawAuthorData.name || "Unknown Author",
    // Convert null values to undefined
    summary: rawAuthorData.summary || undefined,
    twitter: rawAuthorData.twitter || undefined,
    github: rawAuthorData.github || undefined,
    linkedin: rawAuthorData.linkedin || undefined,
  };
}
