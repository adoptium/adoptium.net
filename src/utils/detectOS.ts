export enum UserOS {
  MAC = "MAC",
  WIN = "WIN",
  UNIX = "UNIX",
  LINUX = "LINUX",
  MOBILE = "MOBILE",
  UNKNOWN = "UNKNOWN",
}

export function detectOS(): UserOS {
  let OS = UserOS.UNKNOWN
  if (typeof window !== "undefined") {
    const appVersion = navigator.appVersion;
    if (appVersion.includes("Win")) {
      OS = UserOS.WIN
    } else if (appVersion.includes("Mac")) {
      OS = UserOS.MAC
    } else if (appVersion.includes("X11")) {
      OS = UserOS.UNIX
    } else if (appVersion.includes("Linux")) {
      OS = UserOS.LINUX
    } else if (appVersion.includes("Mobi")) {
      OS = UserOS.MOBILE
    }
  }
  return OS
}
