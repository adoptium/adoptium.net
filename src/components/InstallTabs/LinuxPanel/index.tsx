import React from "react"
import { Link } from "@/i18n/navigation"
import ShellBox from "@/components/ShellBox"
import "../InstallTabs.css"

export const PureLinuxPanel = (): React.ReactNode => {
  // Hardcoded value for latest LTS, should be replaced with actual data fetching in Next.js
  const mostRecentLts = "21"
  return (
    <div>
      <ShellBox>
        <span className="install__text__no-select">
          # Install the latest LTS version (Debian or Ubuntu)
        </span>
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">apt-get install </span>temurin-
        {mostRecentLts}-jdk
      </ShellBox>
      <br />
      <ShellBox>
        <span className="install__text__no-select">
          # Install the latest LTS version (CentOS/RHEL/Fedora)
        </span>
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">yum install </span>temurin-
        {mostRecentLts}-jdk
      </ShellBox>
      <br />
      <ShellBox>
        <span className="install__text__no-select">
          # Install the latest LTS version (openSUSE/SLES)
        </span>
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">zypper install </span>temurin-
        {mostRecentLts}-jdk
      </ShellBox>
      <Link className="install__docs-button" href="/installation/linux">
        Read documentation
      </Link>
    </div>
  )
}

const LinuxPanel = (): React.ReactNode => {
  return <PureLinuxPanel />
}

export default LinuxPanel
