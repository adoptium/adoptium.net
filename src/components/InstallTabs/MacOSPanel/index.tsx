import React from "react"
import ShellBox from "@/components/ShellBox"
import "../InstallTabs.css"

export const PureMacOSPanel = (): React.ReactNode => {
  // Hardcoded value for latest LTS, should be replaced with actual data fetching in Next.js
  const mostRecentLts = "21"
  return (
    <div>
      <ShellBox>
        <span className="install__text__no-select">
          # Install the latest LTS version
        </span>
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">brew install --cask </span>
        temurin@{mostRecentLts}
      </ShellBox>
      <br />
      <ShellBox>
        <span className="install__text__no-select">
          # Install a specific version
        </span>
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">brew install --cask </span>
        temurin@8
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">brew install --cask </span>
        temurin@{mostRecentLts}
      </ShellBox>
      <br />
      <br />
      <br />
      <ShellBox>
        <span className="install__text__no-select">
          # Uninstall a specific version
        </span>
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">brew uninstall --cask </span>
        temurin@{mostRecentLts}
      </ShellBox>
      <a className="install__docs-button" href="https://docs.brew.sh/Manpage">
        Read documentation
      </a>
    </div>
  )
}

const MacOSPanel = (): React.ReactNode => {
  return <PureMacOSPanel />
}

export default MacOSPanel
