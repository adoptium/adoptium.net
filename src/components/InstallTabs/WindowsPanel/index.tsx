import React from "react"
import ShellBox from "@/components/ShellBox"
import "../InstallTabs.css"

export const PureWindowsPanel = (): React.ReactNode => {
  // Hardcoded value for latest LTS, should be replaced with actual data fetching in Next.js
  const mostRecentLts = "21"
  return (
    <div>
      <ShellBox>
        <span className="install__text__no-select">
          # Install the latest version
        </span>
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">winget install </span>
        EclipseAdoptium.Temurin.{mostRecentLts}.JDK
      </ShellBox>
      <br />
      <ShellBox>
        <span className="install__text__no-select">
          # Install a different version
        </span>
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">winget install </span>
        EclipseAdoptium.Temurin.11.JDK
      </ShellBox>
      <br />
      <ShellBox>
        <span className="install__text__no-select">
          # Upgrade the Winget package
        </span>
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">winget upgrade </span>
        EclipseAdoptium.Temurin.{mostRecentLts}.JDK
      </ShellBox>
      <br />
      <ShellBox>
        <span className="install__text__no-select">
          # Uninstall the Winget package
        </span>
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">winget uninstall </span>
        EclipseAdoptium.Temurin.{mostRecentLts}.JDK
      </ShellBox>
      <a
        className="install__docs-button"
        href="https://docs.microsoft.com/en-us/windows/package-manager/winget/"
      >
        Read documentation
      </a>
    </div>
  )
}

const WindowsPanel = (): React.ReactNode => {
  return <PureWindowsPanel />
}

export default WindowsPanel
