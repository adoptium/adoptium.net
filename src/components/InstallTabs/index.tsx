"use client"

import React from "react"
import { Tab, Tabs, TabList, TabPanel } from "react-tabs"
import { detectOS, UserOS } from "@/utils/detectOS"
import "./InstallTabs.css"

import WindowsPanel from "./WindowsPanel"
import MacOSPanel from "./MacOSPanel"
import LinuxPanel from "./LinuxPanel"

const InstallTabs = (): React.ReactNode => {
  const userOS = detectOS()

  const os = {
    win: "Windows (Winget)",
    mac: "macOS (Homebrew)",
    linux: "Linux (RPM/DEB)",
  }

  const installTabSystems: Record<UserOS, string[]> = {
    [UserOS.WIN]: [os.win, os.mac, os.linux],
    [UserOS.MAC]: [os.mac, os.win, os.linux],
    [UserOS.LINUX]: [os.linux, os.mac, os.win],
    [UserOS.UNIX]: [os.linux, os.mac, os.win],
    [UserOS.MOBILE]: [os.win, os.mac, os.linux],
    [UserOS.UNKNOWN]: [os.win, os.mac, os.linux],
  }

  function panelSwitch(): React.ReactNode {
    switch (userOS) {
      case UserOS.MAC:
        return (
          <>
            <TabPanel>
              <MacOSPanel />
            </TabPanel>
            <TabPanel>
              <WindowsPanel />
            </TabPanel>
            <TabPanel>
              <LinuxPanel />
            </TabPanel>
          </>
        )
      case UserOS.LINUX:
      case UserOS.UNIX:
        return (
          <>
            <TabPanel>
              <LinuxPanel />
            </TabPanel>
            <TabPanel>
              <MacOSPanel />
            </TabPanel>
            <TabPanel>
              <WindowsPanel />
            </TabPanel>
          </>
        )
      default:
        return (
          <>
            <TabPanel>
              <WindowsPanel />
            </TabPanel>
            <TabPanel>
              <MacOSPanel />
            </TabPanel>
            <TabPanel>
              <LinuxPanel />
            </TabPanel>
          </>
        )
    }
  }

  return installTabSystems[userOS] !== undefined ? (
    <div className="install">
      <Tabs>
        <div className="install__header">
          <div className="install__header-circles">
            <div className="install__header-grey-circle red" />
            <div className="install__header-grey-circle yellow" />
            <div className="install__header-grey-circle green" />
          </div>
          <div className="install__header-text">
            {userOS === "MAC" ? "zsh" : "bash"}
          </div>
        </div>
        <TabList>
          {installTabSystems[userOS].map((system: string) => (
            <Tab key={system.toString()}>{system}</Tab>
          ))}
        </TabList>
        {panelSwitch()}
      </Tabs>
    </div>
  ) : null
}

export default InstallTabs
