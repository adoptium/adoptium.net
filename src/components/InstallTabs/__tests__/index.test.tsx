import React from "react"
import { render } from "@testing-library/react"
import { describe, expect, it, vi } from "vitest"
import InstallTabs from "../index"
import * as detectOSModule from "@/utils/detectOS"
import { UserOS } from "@/utils/detectOS"

vi.mock("@/utils/detectOS")

describe("Tests for InstallTabs component", () => {
  it("renders correctly", () => {
    vi.spyOn(detectOSModule, "detectOS").mockReturnValue(UserOS.UNKNOWN)
    const { container } = render(<InstallTabs />)
    expect(container).toMatchSnapshot()
  })
  it("renders correctly for macOS", () => {
    vi.spyOn(detectOSModule, "detectOS").mockReturnValue(UserOS.MAC)
    const { container } = render(<InstallTabs />)
    expect(container).toMatchSnapshot()
  })
  it("renders correctly for Linux", () => {
    vi.spyOn(detectOSModule, "detectOS").mockReturnValue(UserOS.LINUX)
    const { container } = render(<InstallTabs />)
    expect(container).toMatchSnapshot()
  })
  it("renders correctly for Unix", () => {
    vi.spyOn(detectOSModule, "detectOS").mockReturnValue(UserOS.UNIX)
    const { container } = render(<InstallTabs />)
    expect(container).toMatchSnapshot()
  })
  it("renders correctly for other", () => {
    vi.spyOn(detectOSModule, "detectOS").mockReturnValue(UserOS.WIN)
    const { container } = render(<InstallTabs />)
    expect(container).toMatchSnapshot()
  })
})
