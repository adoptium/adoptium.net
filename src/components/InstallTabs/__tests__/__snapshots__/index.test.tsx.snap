// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests for InstallTabs component > renders correctly 1`] = `
<div>
  <div
    class="install"
  >
    <div
      class="react-tabs"
      data-rttabs="true"
    >
      <div
        class="install__header"
      >
        <div
          class="install__header-circles"
        >
          <div
            class="install__header-grey-circle red"
          />
          <div
            class="install__header-grey-circle yellow"
          />
          <div
            class="install__header-grey-circle green"
          />
        </div>
        <div
          class="install__header-text"
        >
          bash
        </div>
      </div>
      <ul
        class="react-tabs__tab-list"
        role="tablist"
      >
        <li
          aria-controls="panel_r_0_0"
          aria-disabled="false"
          aria-selected="true"
          class="react-tabs__tab react-tabs__tab--selected"
          data-rttab="true"
          id="tab_r_0_0"
          role="tab"
          tabindex="0"
        >
          Windows (Winget)
        </li>
        <li
          aria-controls="panel_r_0_1"
          aria-disabled="false"
          aria-selected="false"
          class="react-tabs__tab"
          data-rttab="true"
          id="tab_r_0_1"
          role="tab"
        >
          macOS (Homebrew)
        </li>
        <li
          aria-controls="panel_r_0_2"
          aria-disabled="false"
          aria-selected="false"
          class="react-tabs__tab"
          data-rttab="true"
          id="tab_r_0_2"
          role="tab"
        >
          Linux (RPM/DEB)
        </li>
      </ul>
      <div
        aria-labelledby="tab_r_0_0"
        class="react-tabs__tab-panel react-tabs__tab-panel--selected"
        id="panel_r_0_0"
        role="tabpanel"
      >
        <div>
          <pre
            class="shell-box no-highlight"
          >
            <div
              class="shell-box-top"
            />
            <code
              class="shell-box-code"
            >
              <span
                class="install__text__no-select"
              >
                # Install the latest version
              </span>
              <br />
              <span
                class="install__text__no-select"
              >
                $
              </span>
              <span
                class="install-text-command"
              >
                winget install 
              </span>
              EclipseAdoptium.Temurin.
              21
              .JDK
            </code>
          </pre>
          <br />
          <pre
            class="shell-box no-highlight"
          >
            <div
              class="shell-box-top"
            />
            <code
              class="shell-box-code"
            >
              <span
                class="install__text__no-select"
              >
                # Install a different version
              </span>
              <br />
              <span
                class="install__text__no-select"
              >
                $
              </span>
              <span
                class="install-text-command"
              >
                winget install 
              </span>
              EclipseAdoptium.Temurin.11.JDK
            </code>
          </pre>
          <br />
          <pre
            class="shell-box no-highlight"
          >
            <div
              class="shell-box-top"
            />
            <code
              class="shell-box-code"
            >
              <span
                class="install__text__no-select"
              >
                # Upgrade the Winget package
              </span>
              <br />
              <span
                class="install__text__no-select"
              >
                $
              </span>
              <span
                class="install-text-command"
              >
                winget upgrade 
              </span>
              EclipseAdoptium.Temurin.
              21
              .JDK
            </code>
          </pre>
          <br />
          <pre
            class="shell-box no-highlight"
          >
            <div
              class="shell-box-top"
            />
            <code
              class="shell-box-code"
            >
              <span
                class="install__text__no-select"
              >
                # Uninstall the Winget package
              </span>
              <br />
              <span
                class="install__text__no-select"
              >
                $
              </span>
              <span
                class="install-text-command"
              >
                winget uninstall 
              </span>
              EclipseAdoptium.Temurin.
              21
              .JDK
            </code>
          </pre>
          <a
            class="install__docs-button"
            href="https://docs.microsoft.com/en-us/windows/package-manager/winget/"
          >
            Read documentation
          </a>
        </div>
      </div>
      <div
        aria-labelledby="tab_r_0_1"
        class="react-tabs__tab-panel"
        id="panel_r_0_1"
        role="tabpanel"
      />
      <div
        aria-labelledby="tab_r_0_2"
        class="react-tabs__tab-panel"
        id="panel_r_0_2"
        role="tabpanel"
      />
    </div>
  </div>
</div>
`;

exports[`Tests for InstallTabs component > renders correctly for Linux 1`] = `
<div>
  <div
    class="install"
  >
    <div
      class="react-tabs"
      data-rttabs="true"
    >
      <div
        class="install__header"
      >
        <div
          class="install__header-circles"
        >
          <div
            class="install__header-grey-circle red"
          />
          <div
            class="install__header-grey-circle yellow"
          />
          <div
            class="install__header-grey-circle green"
          />
        </div>
        <div
          class="install__header-text"
        >
          bash
        </div>
      </div>
      <ul
        class="react-tabs__tab-list"
        role="tablist"
      >
        <li
          aria-controls="panel_r_2_0"
          aria-disabled="false"
          aria-selected="true"
          class="react-tabs__tab react-tabs__tab--selected"
          data-rttab="true"
          id="tab_r_2_0"
          role="tab"
          tabindex="0"
        >
          Linux (RPM/DEB)
        </li>
        <li
          aria-controls="panel_r_2_1"
          aria-disabled="false"
          aria-selected="false"
          class="react-tabs__tab"
          data-rttab="true"
          id="tab_r_2_1"
          role="tab"
        >
          macOS (Homebrew)
        </li>
        <li
          aria-controls="panel_r_2_2"
          aria-disabled="false"
          aria-selected="false"
          class="react-tabs__tab"
          data-rttab="true"
          id="tab_r_2_2"
          role="tab"
        >
          Windows (Winget)
        </li>
      </ul>
      <div
        aria-labelledby="tab_r_2_0"
        class="react-tabs__tab-panel react-tabs__tab-panel--selected"
        id="panel_r_2_0"
        role="tabpanel"
      >
        <div>
          <pre
            class="shell-box no-highlight"
          >
            <div
              class="shell-box-top"
            />
            <code
              class="shell-box-code"
            >
              <span
                class="install__text__no-select"
              >
                # Install the latest LTS version (Debian or Ubuntu)
              </span>
              <br />
              <span
                class="install__text__no-select"
              >
                $
              </span>
              <span
                class="install-text-command"
              >
                apt-get install 
              </span>
              temurin-
              21
              -jdk
            </code>
          </pre>
          <br />
          <pre
            class="shell-box no-highlight"
          >
            <div
              class="shell-box-top"
            />
            <code
              class="shell-box-code"
            >
              <span
                class="install__text__no-select"
              >
                # Install the latest LTS version (CentOS/RHEL/Fedora)
              </span>
              <br />
              <span
                class="install__text__no-select"
              >
                $
              </span>
              <span
                class="install-text-command"
              >
                yum install 
              </span>
              temurin-
              21
              -jdk
            </code>
          </pre>
          <br />
          <pre
            class="shell-box no-highlight"
          >
            <div
              class="shell-box-top"
            />
            <code
              class="shell-box-code"
            >
              <span
                class="install__text__no-select"
              >
                # Install the latest LTS version (openSUSE/SLES)
              </span>
              <br />
              <span
                class="install__text__no-select"
              >
                $
              </span>
              <span
                class="install-text-command"
              >
                zypper install 
              </span>
              temurin-
              21
              -jdk
            </code>
          </pre>
          <a
            class="install__docs-button"
            href="/installation/linux"
          >
            Read documentation
          </a>
        </div>
      </div>
      <div
        aria-labelledby="tab_r_2_1"
        class="react-tabs__tab-panel"
        id="panel_r_2_1"
        role="tabpanel"
      />
      <div
        aria-labelledby="tab_r_2_2"
        class="react-tabs__tab-panel"
        id="panel_r_2_2"
        role="tabpanel"
      />
    </div>
  </div>
</div>
`;

exports[`Tests for InstallTabs component > renders correctly for Unix 1`] = `
<div>
  <div
    class="install"
  >
    <div
      class="react-tabs"
      data-rttabs="true"
    >
      <div
        class="install__header"
      >
        <div
          class="install__header-circles"
        >
          <div
            class="install__header-grey-circle red"
          />
          <div
            class="install__header-grey-circle yellow"
          />
          <div
            class="install__header-grey-circle green"
          />
        </div>
        <div
          class="install__header-text"
        >
          bash
        </div>
      </div>
      <ul
        class="react-tabs__tab-list"
        role="tablist"
      >
        <li
          aria-controls="panel_r_3_0"
          aria-disabled="false"
          aria-selected="true"
          class="react-tabs__tab react-tabs__tab--selected"
          data-rttab="true"
          id="tab_r_3_0"
          role="tab"
          tabindex="0"
        >
          Linux (RPM/DEB)
        </li>
        <li
          aria-controls="panel_r_3_1"
          aria-disabled="false"
          aria-selected="false"
          class="react-tabs__tab"
          data-rttab="true"
          id="tab_r_3_1"
          role="tab"
        >
          macOS (Homebrew)
        </li>
        <li
          aria-controls="panel_r_3_2"
          aria-disabled="false"
          aria-selected="false"
          class="react-tabs__tab"
          data-rttab="true"
          id="tab_r_3_2"
          role="tab"
        >
          Windows (Winget)
        </li>
      </ul>
      <div
        aria-labelledby="tab_r_3_0"
        class="react-tabs__tab-panel react-tabs__tab-panel--selected"
        id="panel_r_3_0"
        role="tabpanel"
      >
        <div>
          <pre
            class="shell-box no-highlight"
          >
            <div
              class="shell-box-top"
            />
            <code
              class="shell-box-code"
            >
              <span
                class="install__text__no-select"
              >
                # Install the latest LTS version (Debian or Ubuntu)
              </span>
              <br />
              <span
                class="install__text__no-select"
              >
                $
              </span>
              <span
                class="install-text-command"
              >
                apt-get install 
              </span>
              temurin-
              21
              -jdk
            </code>
          </pre>
          <br />
          <pre
            class="shell-box no-highlight"
          >
            <div
              class="shell-box-top"
            />
            <code
              class="shell-box-code"
            >
              <span
                class="install__text__no-select"
              >
                # Install the latest LTS version (CentOS/RHEL/Fedora)
              </span>
              <br />
              <span
                class="install__text__no-select"
              >
                $
              </span>
              <span
                class="install-text-command"
              >
                yum install 
              </span>
              temurin-
              21
              -jdk
            </code>
          </pre>
          <br />
          <pre
            class="shell-box no-highlight"
          >
            <div
              class="shell-box-top"
            />
            <code
              class="shell-box-code"
            >
              <span
                class="install__text__no-select"
              >
                # Install the latest LTS version (openSUSE/SLES)
              </span>
              <br />
              <span
                class="install__text__no-select"
              >
                $
              </span>
              <span
                class="install-text-command"
              >
                zypper install 
              </span>
              temurin-
              21
              -jdk
            </code>
          </pre>
          <a
            class="install__docs-button"
            href="/installation/linux"
          >
            Read documentation
          </a>
        </div>
      </div>
      <div
        aria-labelledby="tab_r_3_1"
        class="react-tabs__tab-panel"
        id="panel_r_3_1"
        role="tabpanel"
      />
      <div
        aria-labelledby="tab_r_3_2"
        class="react-tabs__tab-panel"
        id="panel_r_3_2"
        role="tabpanel"
      />
    </div>
  </div>
</div>
`;

exports[`Tests for InstallTabs component > renders correctly for macOS 1`] = `
<div>
  <div
    class="install"
  >
    <div
      class="react-tabs"
      data-rttabs="true"
    >
      <div
        class="install__header"
      >
        <div
          class="install__header-circles"
        >
          <div
            class="install__header-grey-circle red"
          />
          <div
            class="install__header-grey-circle yellow"
          />
          <div
            class="install__header-grey-circle green"
          />
        </div>
        <div
          class="install__header-text"
        >
          zsh
        </div>
      </div>
      <ul
        class="react-tabs__tab-list"
        role="tablist"
      >
        <li
          aria-controls="panel_r_1_0"
          aria-disabled="false"
          aria-selected="true"
          class="react-tabs__tab react-tabs__tab--selected"
          data-rttab="true"
          id="tab_r_1_0"
          role="tab"
          tabindex="0"
        >
          macOS (Homebrew)
        </li>
        <li
          aria-controls="panel_r_1_1"
          aria-disabled="false"
          aria-selected="false"
          class="react-tabs__tab"
          data-rttab="true"
          id="tab_r_1_1"
          role="tab"
        >
          Windows (Winget)
        </li>
        <li
          aria-controls="panel_r_1_2"
          aria-disabled="false"
          aria-selected="false"
          class="react-tabs__tab"
          data-rttab="true"
          id="tab_r_1_2"
          role="tab"
        >
          Linux (RPM/DEB)
        </li>
      </ul>
      <div
        aria-labelledby="tab_r_1_0"
        class="react-tabs__tab-panel react-tabs__tab-panel--selected"
        id="panel_r_1_0"
        role="tabpanel"
      >
        <div>
          <pre
            class="shell-box no-highlight"
          >
            <div
              class="shell-box-top"
            />
            <code
              class="shell-box-code"
            >
              <span
                class="install__text__no-select"
              >
                # Install the latest LTS version
              </span>
              <br />
              <span
                class="install__text__no-select"
              >
                $
              </span>
              <span
                class="install-text-command"
              >
                brew install --cask 
              </span>
              temurin@
              21
            </code>
          </pre>
          <br />
          <pre
            class="shell-box no-highlight"
          >
            <div
              class="shell-box-top"
            />
            <code
              class="shell-box-code"
            >
              <span
                class="install__text__no-select"
              >
                # Install a specific version
              </span>
              <br />
              <span
                class="install__text__no-select"
              >
                $
              </span>
              <span
                class="install-text-command"
              >
                brew install --cask 
              </span>
              temurin@8
              <br />
              <span
                class="install__text__no-select"
              >
                $
              </span>
              <span
                class="install-text-command"
              >
                brew install --cask 
              </span>
              temurin@
              21
            </code>
          </pre>
          <br />
          <br />
          <br />
          <pre
            class="shell-box no-highlight"
          >
            <div
              class="shell-box-top"
            />
            <code
              class="shell-box-code"
            >
              <span
                class="install__text__no-select"
              >
                # Uninstall a specific version
              </span>
              <br />
              <span
                class="install__text__no-select"
              >
                $
              </span>
              <span
                class="install-text-command"
              >
                brew uninstall --cask 
              </span>
              temurin@
              21
            </code>
          </pre>
          <a
            class="install__docs-button"
            href="https://docs.brew.sh/Manpage"
          >
            Read documentation
          </a>
        </div>
      </div>
      <div
        aria-labelledby="tab_r_1_1"
        class="react-tabs__tab-panel"
        id="panel_r_1_1"
        role="tabpanel"
      />
      <div
        aria-labelledby="tab_r_1_2"
        class="react-tabs__tab-panel"
        id="panel_r_1_2"
        role="tabpanel"
      />
    </div>
  </div>
</div>
`;

exports[`Tests for InstallTabs component > renders correctly for other 1`] = `
<div>
  <div
    class="install"
  >
    <div
      class="react-tabs"
      data-rttabs="true"
    >
      <div
        class="install__header"
      >
        <div
          class="install__header-circles"
        >
          <div
            class="install__header-grey-circle red"
          />
          <div
            class="install__header-grey-circle yellow"
          />
          <div
            class="install__header-grey-circle green"
          />
        </div>
        <div
          class="install__header-text"
        >
          bash
        </div>
      </div>
      <ul
        class="react-tabs__tab-list"
        role="tablist"
      >
        <li
          aria-controls="panel_r_4_0"
          aria-disabled="false"
          aria-selected="true"
          class="react-tabs__tab react-tabs__tab--selected"
          data-rttab="true"
          id="tab_r_4_0"
          role="tab"
          tabindex="0"
        >
          Windows (Winget)
        </li>
        <li
          aria-controls="panel_r_4_1"
          aria-disabled="false"
          aria-selected="false"
          class="react-tabs__tab"
          data-rttab="true"
          id="tab_r_4_1"
          role="tab"
        >
          macOS (Homebrew)
        </li>
        <li
          aria-controls="panel_r_4_2"
          aria-disabled="false"
          aria-selected="false"
          class="react-tabs__tab"
          data-rttab="true"
          id="tab_r_4_2"
          role="tab"
        >
          Linux (RPM/DEB)
        </li>
      </ul>
      <div
        aria-labelledby="tab_r_4_0"
        class="react-tabs__tab-panel react-tabs__tab-panel--selected"
        id="panel_r_4_0"
        role="tabpanel"
      >
        <div>
          <pre
            class="shell-box no-highlight"
          >
            <div
              class="shell-box-top"
            />
            <code
              class="shell-box-code"
            >
              <span
                class="install__text__no-select"
              >
                # Install the latest version
              </span>
              <br />
              <span
                class="install__text__no-select"
              >
                $
              </span>
              <span
                class="install-text-command"
              >
                winget install 
              </span>
              EclipseAdoptium.Temurin.
              21
              .JDK
            </code>
          </pre>
          <br />
          <pre
            class="shell-box no-highlight"
          >
            <div
              class="shell-box-top"
            />
            <code
              class="shell-box-code"
            >
              <span
                class="install__text__no-select"
              >
                # Install a different version
              </span>
              <br />
              <span
                class="install__text__no-select"
              >
                $
              </span>
              <span
                class="install-text-command"
              >
                winget install 
              </span>
              EclipseAdoptium.Temurin.11.JDK
            </code>
          </pre>
          <br />
          <pre
            class="shell-box no-highlight"
          >
            <div
              class="shell-box-top"
            />
            <code
              class="shell-box-code"
            >
              <span
                class="install__text__no-select"
              >
                # Upgrade the Winget package
              </span>
              <br />
              <span
                class="install__text__no-select"
              >
                $
              </span>
              <span
                class="install-text-command"
              >
                winget upgrade 
              </span>
              EclipseAdoptium.Temurin.
              21
              .JDK
            </code>
          </pre>
          <br />
          <pre
            class="shell-box no-highlight"
          >
            <div
              class="shell-box-top"
            />
            <code
              class="shell-box-code"
            >
              <span
                class="install__text__no-select"
              >
                # Uninstall the Winget package
              </span>
              <br />
              <span
                class="install__text__no-select"
              >
                $
              </span>
              <span
                class="install-text-command"
              >
                winget uninstall 
              </span>
              EclipseAdoptium.Temurin.
              21
              .JDK
            </code>
          </pre>
          <a
            class="install__docs-button"
            href="https://docs.microsoft.com/en-us/windows/package-manager/winget/"
          >
            Read documentation
          </a>
        </div>
      </div>
      <div
        aria-labelledby="tab_r_4_1"
        class="react-tabs__tab-panel"
        id="panel_r_4_1"
        role="tabpanel"
      />
      <div
        aria-labelledby="tab_r_4_2"
        class="react-tabs__tab-panel"
        id="panel_r_4_2"
        role="tabpanel"
      />
    </div>
  </div>
</div>
`;
